### PR TITLE
blender: 2.79b -> 2.80

### PR DESCRIPTION
**Pull request by eadwu**
Just transferring stuff from my overlays.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @goibhniu

### CODE REVIEW

File: pkgs/applications/misc/blender/default.nix [hunk 1]
```diff
@@ -14,10 +14,11 @@ with lib;
...
 
 stdenv.mkDerivation rec {
-  name = "blender-2.79b";
+  pname = "blender";
+  version = "2.80";
 
   src = fetchurl {
-    url = "https://download.blender.org/source/${name}.tar.gz";
+    url = "https://download.blender.org/source/${pname}-${version}.tar.gz";
     sha256 = "1g4kcdqmf67srzhi3hkdnr4z1ph4h9sza1pahz38mrj998q4r52c";
```
Comment (eadwu) [inline pkgs/applications/misc/blender/default.nix]: Don't seem to be able to download the file even though it exists?

File: pkgs/applications/misc/blender/default.nix [hunk 2]
```diff
@@ -75,12 +77,6 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  postInstall = optionalString enableNumpy
```
Comment (eadwu) [inline pkgs/applications/misc/blender/default.nix]: Not really sure if `-DPYTHON_NUMPY_PATH` replaces this.

File: pkgs/applications/misc/blender/default.nix [hunk 3]
```diff
@@ -59,6 +60,7 @@ stdenv.mkDerivation rec {
       "-DWITH_PYTHON_INSTALL=OFF"
       "-DWITH_PYTHON_INSTALL_NUMPY=OFF"
     ]
+    ++ optional enableNumpy "-DPYTHON_NUMPY_PATH=${pythonPackages.numpy}/${python.sitePackages}"
```
Comment (eadwu) [inline pkgs/applications/misc/blender/default.nix]: When building from git in the past this seemed to be mandatory, I'll try again when I can download the file.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -17225,8 +17225,8 @@ in
   bleachbit = callPackage ../applications/misc/bleachbit { };
 
   blender = callPackage  ../applications/misc/blender {
-    pythonPackages = python35Packages;
```
Comment (eadwu) [inline pkgs/top-level/all-packages.nix]: Following https://wiki.blender.org/wiki/Building_Blender#Compiler_Versions?

**Comment by worldofpeace** [comment]
We should be able to merge #52818 (maybe out of date now) right?

**Comment by eadwu** [comment]
Should be fine since it matches the required version [1].

[1] https://developer.blender.org/diffusion/B/browse/master/build_files/build_environment/cmake/versions.cmake$116

File: pkgs/applications/misc/blender/default.nix (hunk 3 above)
Comment (eadwu) [reply pkgs/applications/misc/blender/default.nix]:
```
cc1plus: warning: /build/blender/extern/audaspace/PYTHON_NUMPY_PATH-NOTFOUND/numpy/core/include: No such file or directory [-Wmissing-include-dirs]
[ 91%] Building C object source/blender/makesrna/intern/CMakeFiles/makesrna.dir/rna_boid.c.o
/build/blender/extern/audaspace/bindings/python/PySound.cpp:70:10: fatal error: numpy/ndarrayobject.h: No such file or directory
 #include <numpy/ndarrayobject.h>
          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [extern/audaspace/CMakeFiles/audaspace-py.dir/build.make:154: extern/audaspace/CMakeFiles/audaspace-py.dir/bindings/python/PySound.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2594: extern/audaspace/CMakeFiles/audaspace-py.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by eyJhb** [comment]
@eadwu shouldn't we just fetchfromgit and use tags instead? For me it seems like the best way, also because the mirrors are VERY slow compared to git :)

File: pkgs/applications/misc/blender/default.nix (hunk 1 above)
Comment (jansol) [reply pkgs/applications/misc/blender/default.nix]: Looks okay here, probably just blender.org being hammered due to the release announcement. What with them being at SIGGRAPH to show it off and all ^.^

**Comment by ashkitten** [comment]
so i merged this pr into my nixpkgs and got this...
image

**Comment by eadwu** [comment]
Hash hasn't been calculated.

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jansol) [reply pkgs/top-level/all-packages.nix]: Doesn't seem like there is any particular reason to stick to gcc6 (did blender recommend it for 2.79?)

Comment (jansol) [reply pkgs/top-level/all-packages.nix]: Maybe add a comment pointing to that page for future reference.

**Comment by ar1a** [comment]
that was quick!

**Comment by eadwu** [comment]
Well still haven't been able to download the file, although I have reached the `1.0 MiB` mark. Through the git source it builds fine with opencollada `1.6.68`. Derivation output was `/nix/store/fkvwfjsbgs5wqicy3v1i3xqjmbr1wmv0-blender-2.80`.

**Comment by eyJhb** [comment]
@eadwu I got it yesterday, I have put it on my server, feel free to grab it! http://srtutti.zurlius.dk:8394/blender-2.80.tar.gz

File: pkgs/applications/misc/blender/default.nix [hunk 5]
```diff
@@ -58,13 +57,10 @@ stdenv.mkDerivation rec {
       "-DPYTHON_VERSION=${python.pythonVersion}"
       "-DWITH_PYTHON_INSTALL=OFF"
       "-DWITH_PYTHON_INSTALL_NUMPY=OFF"
+      "-DPYTHON_NUMPY_PATH=${pythonPackages.numpy}/${python.sitePackages}"
     ]
     ++ optional jackaudioSupport "-DWITH_JACK=ON"
-    ++ optionals cudaSupport
-      [ "-DWITH_CYCLES_CUDA_BINARIES=ON"
-        # Disable architectures before sm_30 to support new CUDA toolkits.
-        "-DCYCLES_CUDA_BINARIES_ARCH=sm_30;sm_35;sm_37;sm_50;sm_52;sm_60;sm_61"
```
Comment (eadwu) [inline pkgs/applications/misc/blender/default.nix]: Seems like it isn't needed anymore [1].

[1] https://git.blender.org/gitweb/gitweb.cgi/blender.git/blob/v2.80:/CMakeLists.txt#l410

**Comment by eyJhb** [comment]
@eadwu  had to put it inside a container, as I was unsure if SimpleHTTPServer is safe. So if your download was interrupted, then restart it :)

**Comment by eadwu** [comment]
Downloaded it beforehand and it matched the hash so it's fine to me. Anyway with the source provided by @eyJhb it also compiles fine with opencollada `1.6.68`, `/nix/store/sbyd66xd3ygqb9awr6j55k2nhmaf3bnx-blender-2.80`.

**Comment by eyJhb** [comment]
@eadwu is there any reason we are not cloning from Git?

**Comment by eadwu** [comment]
Not sure, but I don't like changing the source unless it's missing for 2 or more days.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -17227,8 +17227,10 @@ in
...
-    stdenv = gcc6Stdenv;
+    # https://developer.blender.org/diffusion/B/browse/master/build_files/build_environment/cmake/versions.cmake;v2.80$146
+    pythonPackages = python37Packages;
+    # https://wiki.blender.org/wiki/Building_Blender#Compiler_Versions
+    stdenv = gcc7Stdenv;
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: Since Python 3.7 and GCC 7 are the default these lines can be removed.

**Comment by edolstra** [comment]
Some changes from my blender-2.80 Nix expression that you may want to copy:
```
diff --git a/pkgs/applications/misc/blender/default.nix b/pkgs/applications/misc/blender/default.nix
index 353cf620ee7..6d9aa7a4384 100644
--- a/pkgs/applications/misc/blender/default.nix
+++ b/pkgs/applications/misc/blender/default.nix
@@ -46,10 +57,7 @@ stdenv.mkDerivation rec {
       "-DWITH_INSTALL_PORTABLE=OFF"
       "-DWITH_FFTW3=ON"
       #"-DWITH_SDL=ON"
-      "-DWITH_GAMEENGINE=ON"
       "-DWITH_OPENCOLORIO=ON"
-      "-DWITH_SYSTEM_OPENJPEG=ON"
-      "-DWITH_PLAYER=ON"
       "-DWITH_OPENSUBDIV=ON"
       "-DPYTHON_LIBRARY=${python.libPrefix}m"
       "-DPYTHON_LIBPATH=${python}/lib"
```

Also I did still need the `wrapProgram` to set the `PYTHONPATH` to `numpy` but that was a couple of months ago.

**Comment by eadwu** [comment]
Verified the hash since I can actually download the source now. Added back the wrapper since `blender --python-expr 'import numpy'` doesn't work without it.

**Comment by jansol** [comment]
Is there anything left blocking this?

**Comment by eadwu** [comment]
Don't think so.

**Pull request merged**